File: python/bubble_sort.py
from math import nan


sequence = [4, 3, 5, 0, 1]


def bubble_sort(array):
    swaps = 0
    previous = nan
    for i, num in enumerate(array):
        if i == 0:
            previous = num
            continue
        elif num < previous:
            smaller_num = array.pop(i)
            if (i-2) < 0:
                array.insert((i-1), smaller_num)
                swaps += 1
                previous = num
                bubble_sort(array)
            else:
                array.insert((i-2), smaller_num)
                swaps += 1
                previous = num
                bubble_sort(array)
        elif num > previous:
            previous = num
            continue

    return(array)

print(bubble_sort(sequence))

File: js/bubbleSort.js
function bubbleSort(sequence) {
  let count = 0
  for (let i = 0; i < sequence.length; i++) {
    for (let j = 0; j < sequence.length - i - 1; j++) {
      if (sequence[j] > sequence[j + 1]) {
        //let currentItem = sequence[j]
        //let nextItem = sequence[j + 1]
        let current = sequence[j]
        sequence[j] = sequence[j + 1]
        sequence[j + 1] = current
        count++
      }
    }
  }
  return sequence, count
}


console.log((bubbleSort([1, 7, 8, 6, 3, 2, 4])))